cmd/swarm/mru.go

```diff
@@ -0,0 +1,117 @@
+// Copyright 2016 The go-ethereum Authors
+// This file is part of go-ethereum.
+//
+// go-ethereum is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// go-ethereum is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with go-ethereum. If not, see <http://www.gnu.org/licenses/>.
+
+// Command resource allows the user to create and update signed mutable resource updates
+package main
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/ethereum/go-ethereum/common/hexutil"
+
+	"github.com/ethereum/go-ethereum/cmd/utils"
+	swarm "github.com/ethereum/go-ethereum/swarm/api/client"
+	"github.com/ethereum/go-ethereum/swarm/storage/mru"
+	"gopkg.in/urfave/cli.v1"
+)
+
+// swarm resource [--raw] create <name> <frequency> <0x Hexdata>
+// swarm resource update <Manifest Address or ENS domain> <0x Hexdata>
```

Comment:
Yes. What can be confusing is that, for the cli user, the create command actually creates a resource *and* sends off the first update in one go. I was hoping to simplify the interaction with this strategy.

Another approach would be to have a `create` command to create a resource which just returns the manifest hash and then the user has to invoke `update` to send the first update, otherwise the resource wouldn't have any content.

I am fine either way. The second approach is "cleaner": one command creates, other command updates, but also more complicated for a user. What do you think?

Reply:
> Another approach would be to have a create command to create a resource which just returns the manifest hash and then the user has to invoke update to send the first update, otherwise the resource wouldn't have any content

Yeah I remember pondering this too, and I too think it's cleaner. Let's go for that.

Reply:
OK, I'll implement this directly in #704.

Reply:
@nolash Finished modifying the CLI to decouple resource creation from updating according to these comments. The changes are in the last two commits of #704.
Creating a resource: `swarm resource create <frequency> [--name <name>] [--data <0x Hexdata> [--rawmru]]`
Updating a resource: `swarm resource update <Manifest Address or ENS domain> <0x Hexdata> [--rawmru]`
as you can see from the syntax, adding data during resource creation is now optional, but kept as a convenient way to create & initialize in one command.